traffic_ops/traffic_ops_golang/api/api.go

```diff
@@ -134,6 +134,7 @@ func handleSimpleErr(w http.ResponseWriter, r *http.Request, statusCode int, use
 		return
 	}
 	*r = *r.WithContext(context.WithValue(r.Context(), tc.StatusKey, statusCode))
+	w.WriteHeader(statusCode)
```

Comment:
Is this necessary for the 500 to be returned properly?

The wrapper stuff is supposed to require the line immediately above it,
`*r = *r.WithContext(context.WithValue(r.Context(), tc.StatusKey, statusCode))`

(Something I'd like to fix, but it is what it is today). 

If the `WriteHeader` doesn't actually do anything, it should be removed. If it does, we should figure out what changed, and if we can remove the StatusKey Context stuff.

Reply:
So the `gzipResponse` function appears to handle writing the status header. I added this line because my added unit test was failing, but I've updated the unit test to also wrap with `wrapHeaders` which includes `gzipResponse`.